frontend/src/models/fms_api_models.ts

```diff
@@ -0,0 +1,106 @@
+/**
```

Comment:
If you want to introduce a structure for models in the frontend maybe you should move the models.js file here. Maybe rename it to backend_models... The idea is not to introduce new varying way to deal with models without integrating the existing code base. In general try to separate effort to overhaul the code from feature creation. That way we don't get many small departure from the way that things were done all over the code (even if your part is nice and shiny it makes the code that much harder to track and understand if the overhaul is not complete).